--- libs/boost/build-android.sh	2023-08-15 17:57:20.714121402 +0800
+++ libs/boost/build-android.sh	2023-08-15 11:55:10.988665000 +0800
@@ -402,7 +402,7 @@
 		TOOLSET=clang
 		CONFIG_VARIANT=ndk19
 		;;
-	"22.1"|"23.0"|"23.1"|"25.0"|"25.1"|"25.2")
+	"22.1"|"23.0"|"23.1"|"23.2"|"25.0"|"25.1"|"25.2")
 		TOOLCHAIN=${TOOLCHAIN:-llvm}
 		CXXPATH=$AndroidNDKRoot/toolchains/${TOOLCHAIN}/prebuilt/${PlatformOS}-x86_64/bin/clang++
 		TOOLSET=clang
@@ -423,7 +423,7 @@
 
     case "$NDK_RN" in
       # NDK 17+: Support for ARMv5 (armeabi), MIPS, and MIPS64 has been removed.
-      "17.1"|"17.2"|"18.0"|"18.1"|"19.0"|"19.1"|"19.2"|"20.0"|"20.1"|"21.0"|"21.1"|"21.2"|"21.3"|"21.4"|"22.1"|"23.0"|"23.1"|"25.0"|"25.1"|"25.2")
+      "17.1"|"17.2"|"18.0"|"18.1"|"19.0"|"19.1"|"19.2"|"20.0"|"20.1"|"21.0"|"21.1"|"21.2"|"21.3"|"21.4"|"22.1"|"23.0"|"23.1"|"23.2"|"25.0"|"25.1"|"25.2")
         ARCHLIST="arm64-v8a armeabi-v7a x86 x86_64"
         ;;
       *)
