--- libs/sdl/src/video/x11/SDL_x11sym.h	2012-01-19 14:30:06.000000000 +0800
+++ libs/sdl/src/video/x11/SDL_x11sym2.h	2023-08-15 19:56:47.159676037 +0800
@@ -163,10 +163,18 @@
  * Not required...these only exist in code in headers on some 64-bit platforms,
  *  and are removed via macros elsewhere, so it's safe for them to be missing.
  */
+#define SDL_VIDEO_DRIVER_X11_CONST_PARAM_XDATA32 1
 #ifdef LONG64
 SDL_X11_MODULE(IO_32BIT)
-SDL_X11_SYM(int,_XData32,(Display *dpy,register long *data,unsigned len),(dpy,data,len),return)
-SDL_X11_SYM(void,_XRead32,(Display *dpy,register long *data,long len),(dpy,data,len),)
+#if SDL_VIDEO_DRIVER_X11_CONST_PARAM_XDATA32
+SDL_X11_SYM(int,_XData32,(Display *dpy,register _Xconst long *data,unsigned
+len),(dpy,data,len),return)
+#else
+SDL_X11_SYM(int,_XData32,(Display *dpy,register long *data,unsigned len),
+(dpy,data,len),return)
+#endif
+SDL_X11_SYM(void,_XRead32,(Display *dpy,register long *data,long len),
+(dpy,data,len),)
 #endif
 
 /*
